django/db/models/base.py

```diff
@@ -878,7 +878,8 @@ def _get_next_or_previous_in_order(self, is_next):
     def prepare_database_save(self, field):
         if self.pk is None:
             raise ValueError("Unsaved model instance %r cannot be used in an ORM query." % self)
-        return getattr(self, field.remote_field.field_name)
+        val = getattr(self, field.remote_field.field_name)
```

Comment:
Maybe the problem is that field_name refers to field.name when field.attname should be used. I'm not sure how to get field.attname.

It seems val.pk is wrong for two reasons, first the relation might be pointing to some other field than pk, and also it might be that retrieving the val instance might require executing a queryset.